src/libraries/icubmod/usbCamera/linux/V4L_camera.cpp

```diff
@@ -694,28 +706,44 @@ bool V4L_camera::close()
 bool V4L_camera::getRgbBuffer(unsigned char *buffer)
 {
     mutex.wait();
-    imageProcess(param.raw_image);
-//     memcpy(buffer, param.dst_image, param.rgb_src_image_size*12/16);
-//     memcpy(buffer, param.tmp_image2, param.rgb_src_image_size*12/16);
-//     memcpy(buffer, param.outMat.data, param.src_fmt.fmt.pix.width * param.src_fmt.fmt.pix.height * 3);
-
-//     memcpy(buffer, param.img.data, param.src_fmt.fmt.pix.width*12/16 * param.src_fmt.fmt.pix.height * 3);
-//     std::cout << "dst_image_size"<< param.dst_image_size << "; compute " << param.src_fmt.fmt.pix.width * param.src_fmt.fmt.pix.height * 3 *12 /16  << std::endl;
-//     memcpy(buffer, param.outMat.data, param.src_fmt.fmt.pix.width * param.src_fmt.fmt.pix.height * 3 *12 /16 );
-//     int __size = param.outMat.total();
-    memcpy(buffer, param.outMat.data, param.outMat.total()*3);
-    mutex.post();
-    return true;
+    if(configured){
+        imageProcess(param.raw_image);
+    //     memcpy(buffer, param.dst_image, param.rgb_src_image_size*12/16);
+    //     memcpy(buffer, param.tmp_image2, param.rgb_src_image_size*12/16);
+    //     memcpy(buffer, param.outMat.data, param.src_fmt.fmt.pix.width * param.src_fmt.fmt.pix.height * 3);
+
+    //     memcpy(buffer, param.img.data, param.src_fmt.fmt.pix.width*12/16 * param.src_fmt.fmt.pix.height * 3);
+    //     std::cout << "dst_image_size"<< param.dst_image_size << "; compute " << param.src_fmt.fmt.pix.width * param.src_fmt.fmt.pix.height * 3 *12 /16  << std::endl;
+    //     memcpy(buffer, param.outMat.data, param.src_fmt.fmt.pix.width * param.src_fmt.fmt.pix.height * 3 *12 /16 );
+    //     int __size = param.outMat.total();
+        memcpy(buffer, param.outMat.data, param.outMat.total()*3);
+        mutex.post();
+        return true;
+    }
+    else
+    {
+        yError()<<"usbCamera: unable to get the buffer, device unitialized";
+        mutex.post();
+        return false;
+    }
 }
 
 // IFrameGrabber Interface
 bool V4L_camera::getRawBuffer(unsigned char *buffer)
 {
+
     mutex.wait();
-    imageProcess(param.raw_image);
-    memcpy(buffer, param.dst_image, param.dst_image_size);
-    mutex.post();
-    return true;
+    if(configured){
+        imageProcess(param.raw_image);
+        memcpy(buffer, param.dst_image, param.dst_image_size);
+        mutex.post();
+        return true;
+    }
+    else{
+        yError()<<"usbCamera: unable to get the buffer, device unitialized";
+        mutex.post();
+        return false;
+    }
 }
```

Comment:
I checked with clang it appears this warning 
`/icub-main/src/libraries/icubmod/usbCamera/linux/V4L_camera.cpp:1818:1: warning: control may reach end of non-void function [-Wreturn-type]
}`